src/lib/network/clients/save_network.rb

```diff
@@ -0,0 +1,287 @@
+require "y2storage"
+require "network/install_inf_convertor"
+require "network/network_autoconfiguration"
+require "network/network_autoyast"
+
+module Yast
+  class SaveNetworkClient < Client
+    include Logger
+
+    def main
+      textdomain "network"
+
+      Yast.import "DNS"
+      Yast.import "FileUtils"
+      Yast.import "NetworkStorage"
+      Yast.import "Installation"
+      Yast.import "String"
+      Yast.import "Mode"
+      Yast.import "Arch"
+
+      Yast.include self, "network/routines.rb"
+      Yast.include self, "network/complex.rb"
+
+      # for update system don't copy network from inst_sys (#325738)
+      if !Mode.update
+        save_network
+      else
+        Builtins.y2milestone("update - skip save_network")
+      end
+      # EOF
+
+      nil
+    end
+
+    def adjust_for_network_disks(file)
+      # storage-ng
+      # Check if installation is targeted to a remote destination.
+      devicegraph = Y2Storage::StorageManager.instance.y2storage_staging
+      is_disk_in_network = devicegraph.filesystem_in_network?(
+        Installation.destdir
+      )
+
+      return unless is_disk_in_network
+
+      log.info("Directory \"#{Installation.destdir}\" is on a network based device")
```

Comment:
The old code looked like this (with the log message before the return):
```ruby
log.info("Network based device: #{remote_access}")
return if remote_access == :no
```

Of course, the new log message is nicer. But the old code always wrote something to the logs (no matter if the device is in network of not). That's better because it allows us to know that the check was performed.